src/main/java/ai/mindgard/MindgardSettingsUI.java

```diff
@@ -139,6 +134,23 @@ public MindgardSettingsUI() {
             }
         }
 
+        this.uriManager = new MindgardServerConfiguration(settings);
+
+        if (settings != null) {
+            selector = Optional.ofNullable(settings.selector()).orElse(selector);
+            projectID = Optional.ofNullable(settings.projectID()).orElse(projectID);
+            dataset = Optional.ofNullable(settings.dataset()).orElse(dataset);
+            systemPrompt = Optional.ofNullable(settings.systemPrompt()).orElse(systemPrompt);
+            customDataset = Optional.ofNullable(settings.customDatasetFilename()).map(File::new).orElse(customDataset);
+            exclude = Optional.ofNullable(settings.exclude()).orElse(exclude);
+            include = Optional.ofNullable(settings.include()).orElse(include);
+            promptRepeats = Optional.ofNullable(settings.promptRepeats()).orElse(DEFAULT_PROMPT_REPEATS);
+            parallelism = Optional.ofNullable(settings.parallelism()).orElse(DEFAULT_PARALLELISM);
```

Comment:
potentially not needed to refactor if we don't need to instantiate uriManager and can just directly use the settings object